src/hooks/useFetchMovies.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { MovieInfo } from '../components/movie-tile/movie-tyle.types';
+
+export const useFetchMovies = (
+  searchQuery: string,
+  searchBy: 'title' | 'genres' = 'title',
+  sortBy: 'title' | 'release_date' = 'release_date',
+  genreFilter?: string,
+) => {
+  const [movies, setMovies] = useState<MovieInfo[]>([]);
+
+  useEffect(() => {
+    const fetchMovies = async () => {
+      try {
+        let url = 'http://localhost:4000/movies';
+        const params: Record<string, string> = {};
+        if (searchQuery) {
+          params.search = encodeURIComponent(searchQuery);
+          params.searchBy = searchBy;
+        }
+        if (genreFilter) {
+          params.filter = genreFilter;
+        }
+        params.sortBy = sortBy;
```

Comment:
I tried to sort movies by title locally... looks like it doesn't work. I guess you have to add &sortOrder=asc to the query string.

Reply:
Updated.